src/checker.rs

```diff
@@ -1,87 +1,143 @@
-use anyhow::{anyhow, Result};
+use anyhow::{anyhow, bail, Result};
 use log::warn;
 use std::fmt;
-use std::process::{Command, Output};
+use std::process::Command;
 
-fn check_cmd(program: &str, arg: &str) -> Result<Output> {
-    Command::new(program).arg(arg).output().map_err(Into::into)
+struct BinDep {
+    program: String,
+    installed: bool,
+    version: Option<Version>,
+    required_version: Option<Version>,
+}
+
+impl BinDep {
+    fn build(
+        program: &str,
+        arg: &'static str,
+        version_prefix: Option<&'static str>,
+        required_version: Option<Version>,
+    ) -> Result<Self> {
+        let output = Command::new(program).arg(arg).output()?;
```

Comment:
This will exit with just `error: No such file or directory (os error 2)` if the program is not found.